[CS2113-F13-2] BrokeMan

namsengi11 opened the pull request:
BrokeMan helps students running on low budget by organizing their budget usage and expenditures. It provides various features to help students minimize their spending and raise awareness on their spendings. It is optimized for CLI users so that budget and expenditure recording can be done quicker through entering commands.

Samueltansw committed:
fixed checkstyle

Samueltansw committed:
Merge pull request #62 from Samueltansw/master

updated MESSAGE_USAGE in command classes

SpeciLiam committed:
Added Save component for Budget and Read Component

SpeciLiam committed:
Added Save component for Budget and Read Component

SpeciLiam committed:
CheckStyle

SpeciLiam committed:
Merge pull request #63 from AY2223S2-CS2113-F13-2/master

Added Save component for Budget and Read Component

sistine-yu committed:
Update the command list of User Guide

jingyaaa commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,253 @@
+
+![Architecture Diagram](images/ArchitectureDiagram.png)
+
+The ***architecture diagram*** given above is explains the high level design of the program.
```
There seems to be a typo in this sentence

jingyaaa commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,253 @@
+
+---
+
+|Version| As a ... | I want to ...                                                         | So that I can ...                                |
```
User stories are repeated in the DG, there are 2 tables

jingyaaa commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,253 @@
+
+---
+
+## implementation
```
"i" is in lowercase

denzelcjy replied:
Can consider to change it to "Implementation" instead

jingyaaa commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,253 @@
 # Developer Guide
```
Perhaps can add more diagrams of different types

jingyaaa commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,253 @@
+
+## Design
+
+### Architecture
```
Architecture diagram is clear and well explained

denzelcjy commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,253 @@
+   - [UI component](#ui-component)
+   - [Parser component](#parser-component)
+   - [Storage component](#storage-component)
+   - [Common classes](#common-classes)
```
The link here for the common classes does not seem to be working, perhaps edit the content in the link bracket to be #common-class for it to work?

denzelcjy commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,253 @@
+   - [Storage component](#storage-component)
+   - [Common classes](#common-classes)
+3. [Implementation](#implementation)
+4. [Appendix: Requirements](#appendix--requirements)
+5. [Appendix: Instructions for manual testing](#appendix--instructions-for-manual-testing)
```
These 2 links also do not seem to work when I clicked it, perhaps consider changing the '--' to '-' after 'appendix'?

denzelcjy commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,253 @@
 
+## Acknowledgements
+x
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
+original source as well}
```
there seems to be duplicate sentences here, can consider updating and filling it up whenever possible!

denzelcjy commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,253 @@
+
+### Architecture
+
+![Architecture Diagram](images/ArchitectureDiagram.png)
+
+The ***architecture diagram*** given above is explains the high level design of the program.
+
+Given below is a quick overview of the main components and how they interact with each other.
+
+**Main components of architecture**
+
+`BrokeMan` has one class [`Main`](https://github.com/AY2223S2-CS2113-F13-2/tp/blob/master/src/main/java/seedu/brokeMan/BrokeMan.java), which is responsible for:
+- At program launch: Initialises the components in the correct sequence, and connect them up with each other
+- At program termination: Shuts down the components and invokes cleanup methods where necessary.
+
+[`Common`](#common-class) represents a collection of messages used by multiple other components.
```
![image](https://user-images.githubusercontent.com/88487773/228703886-4b2fd0c7-095e-4de3-9560-4cfa7de2110b.png)
A little confused with the architecture diagram, in line 46 it mentions that "common represents a collection of messages used by multiple other components, but there seems to be no arrow associated with the "common" component in the diagram, is it supposed to be intended this way?

denzelcjy commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,253 @@
+
+## implementation
+
+### Entry
+
+Entry class is the underlying superclass for Expense and Income classes. It establishes the common attributes and
+methods that are necessary to represent Expenses and Incomes. Abstract class is used to represent their common features
+to maximize code reusability and increase maintainability.
+
+Private attributes
+
+Info: String that stores the description of the entry
+
+Amount: Double that stores the monetary value of entry
+
+Time: LocalDateTime that stores the date and time of entry
+
+Category: Category that stores the type tag of entry
+
+**Methods**
+
+Getters can be used to provide the private attributes to other classes
+
+editDescription(), editAmount(), editTime(), editCategory()
+
+* Takes in corresponding parameters to edit the private attributes.
+* Used by EntryList to make edits
+
+isSameMonth()
```
Nice work having a short summary as a description, can consider bolding "Private attributes" to make it aligned in formatting with the "Methods" that you already bolded below. 
Can also consider adding bullets to indent the subpoints "Info, Amount, Time, Category" to increase readability!
Alternatively, maybe you could use a table as another way to display these details :)

denzelcjy commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,253 @@
+
+Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unlikely to have) - `*`
+
+| Priority | Version | As a ... | I want to ...                             | So that I can ...                                |
+|----------|---------|----------|-------------------------------------------|--------------------------------------------------|
+| `* * *`  | v1.0    | user     | add, delete, edit, and list my income     | manage my income                                 |
+| `* * *`  | v1.0    | user     | add, delete, edit, and list my expenses   | manage my expenses                               |
+| `* * *`  | v1.0    | user     | set and view my budget                    | set expectation of how much money I should use   |
+| `* * *`  | v1.0    | user     | view how much of the budget I spend       | manage and change my spending habit as necessary |
+| `* * *`  | v1.0    | user     | view all command that I can enter         | get help on the features if necessary            |
+| `* * *`  | v2.0    | user     | list monthly expenses, income, and budget | refer to financial status in previous months     |
+| `* * *`  | v2.0    | user     | save all my income and expenses entered   | so that I can refer to it next time I return     |
```
This is a second User Stories table that is almost similar to the first one above, is this supposed to be combined or meant to be an additional table? Can consider providing just 1 table to reduce confusion 👍

denzelcjy commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,253 @@
+### SaveExpense, SaveIncome
+
+The SaveExpense and SaveIncome class deal with saving in the user inputted data locally so that it can be later accessed.
+**Methods**
```
![image](https://user-images.githubusercontent.com/88487773/228706632-3fcf6a6e-b17f-44bf-8376-6fcd7fcfbb32.png)
Can consider adding a line break here so that "Methods" is on a new line

denzelcjy commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,253 @@
+
+* Takes in a list of entries and returns the sum of all entry's monetary value.
+
+SortEntryByAmount()
+
+* Returns a sorted list of entry by amount of entry.
+
+SortEntryByDate()
```
are these 2 methods meant to be written as `sortEntryByAmount` and `sortEntryByDate()` instead? To follow the camelCase convention for method naming

denzelcjy reviewed (commented):
Nice work on trying to segment the information for better readability, can consider using more bullets, whitespaces and different font sizes in order to keep it more consistent. Can consider adding sequence diagrams, and maybe a class diagram to show how different classes interact with each other. Overall good job! 👍

sistine-yu committed:
Update the command cheatsheet of User Guide

sistine-yu committed:
Update the command cheatsheet of User Guide

sistine-yu committed:
Update the text ui

sistine-yu committed:
Merge branch 'master' of https://github.com/sistine-yu/tp
Merge an updated upstream

sistine-yu committed:
Merge pull request #65 from sistine-yu/master

UG update

Samueltansw committed:
updated UserGuide.md and enabled assertions in build.gradle

Samueltansw committed:
added table of contents for user guide

Mnsd05 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,251 @@
+## Design
+
+### Architecture
+
+![Architecture Diagram](images/ArchitectureDiagram.png)
+
+The ***architecture diagram*** given above is explains the high level design of the program.
```
You may add the Main class to the diagram

Mnsd05 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,251 @@
+
+### EntryList component
+
+### Common class
```
This part is empty, you might want to delete or add the sequence diagrams for each component

Mnsd05 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,251 @@
+
+---
 
-## Instructions for manual testing
+## Appendix: Instructions for manual testing
 
 {Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
```
You can try to include how you give some input to the program and how the program should respond. For example:

Input: abc

Expected: Oops, I don't know what that means

Mnsd05 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,251 @@
+| `* * *`  | v1.0    | user     | view how much of the budget I spend       | manage and change my spending habit as necessary |
+| `* * *`  | v1.0    | user     | view all command that I can enter         | get help on the features if necessary            |
+| `* * *`  | v2.0    | user     | list monthly expenses, income, and budget | refer to financial status in previous months     |
+| `* * *`  | v2.0    | user     | save all my income and expenses entered   | so that I can refer to it next time I return     |
```
Well done! I find that priority is a good feature of your user stories as it gives an idea of the most important features in your product

Mnsd05 reviewed (commented):
Overall, your developer guide is easy to understand, well done!

Samueltansw committed:
Merge pull request #66 from Samueltansw/master

edited user guide

SpeciLiam committed:
Developer Guide for Save  Methods and Classes.

SpeciLiam committed:
Merge pull request #67 from AY2223S2-CS2113-F13-2/master

Developer Guide for Save  Methods and Classes.

sistine-yu committed:
Update DG miscellaneous

sistine-yu committed:
merge the updated upstreamMerge branch 'master' of https://github.com/sistine-yu/tp

namsengi11 committed:
Add Entry Test

Added multiple J Unit test classes to test EntryList, ExpenseList, and IncomeList.

namsengi11 committed:
Resolve checkstyle issue

namsengi11 committed:
Resolve checkstyle issue

namsengi11 committed:
Resolve checkstyle issue

namsengi11 committed:
Resolve checkstyle issue

namsengi11 committed:
Resolve contrasting save file

namsengi11 committed:
Merge pull request #69 from namsengi11/test

Add Entry Test

Samueltansw committed:
added assertions to parser class and some minor changes to UserInput

Samueltansw committed:
Merge remote-tracking branch 'upstream/master' into branch-AddAssertions

Samueltansw committed:
fixed indentation error

Samueltansw committed:
fixed indentation error

namsengi11 committed:
Resolve version conflict issue

Samueltansw committed:
removed wrong assertion in parser class

Samueltansw committed:
updated expected.txt

Samueltansw committed:
Merge pull request #70 from Samueltansw/branch-AddAssertions

add assertions

Samueltansw committed:
updated help message

Samueltansw committed:
Merge pull request #71 from Samueltansw/master

updated help message

Samueltansw committed:
Merge pull request #68 from sistine-yu/master

Update DG miscellaneous

namsengi11 committed:
Update UserGuide.md

Samueltansw committed:
added architecture sequence diagram to developer guide

Samueltansw committed:
Merge remote-tracking branch 'upstream/master'

Samueltansw committed:
Merge pull request #72 from Samueltansw/master

added architecture sequence diagram to developer guide

sistine-yu committed:
Update UG hyperlink

sistine-yu committed:
Merge branch 'AY2223S2-CS2113-F13-2:master' into master

sistine-yu committed:
Merge pull request #73 from sistine-yu/master

Update UG hyperlink

Samueltansw committed:
updated developerGuide.md

Samueltansw committed:
Merge remote-tracking branch 'upstream/master'

Samueltansw committed:
minor edit tod developer guide

Samueltansw committed:
Merge pull request #74 from Samueltansw/master

Edited developer guide

Samueltansw committed:
added example outputs that the user will see based on the inputs they entered

Samueltansw committed:
Merge pull request #75 from Samueltansw/master

added example outputs that the user will see based on the inputs they entered

Samueltansw committed:
minor edits to user guide

Samueltansw committed:
Merge pull request #76 from Samueltansw/master

minor edits to user guide

jeyvia commented on src/main/java/seedu/brokeMan/parser/StringToTime.java:
```diff
@@ -0,0 +1,71 @@
+        } else if (!areWordsValidTimeNumbers(timeByWord)) {
+            throw new DateTimeException(MESSAGE_INVALID_TIME);
+        }
+        // dummy assert
```
Do remove such comments before the final submission!

jeyvia commented on src/main/java/seedu/brokeMan/entry/EntryList.java:
```diff
@@ -0,0 +1,171 @@
+     */
+    public static void listEntry(List<Entry> entryList) {
+        if (entryList.size() == 0) {
+            Ui.showToUser("The requested list is empty\n|");
```
Please do try to minimise the use of magic strings! Especially since you have a common class already :)

jeyvia commented on src/main/java/seedu/brokeMan/parser/Parser.java:
```diff
@@ -0,0 +1,557 @@
+     *
+     * @param userFullInput full line of user input entered by user
+     * @return Command class based on the user's input
+     */
```
Well done! The parser is well-implemented with no long methods.

jeyvia commented on src/main/java/seedu/brokeMan/command/ExitCommand.java:
```diff
@@ -0,0 +1,25 @@
+            "|  Example: " + COMMAND_WORD;
+    public void execute() {
+        Ui.showToUserWithLineBreak("Exiting program...", "");
+        SaveExpense.writeFile(expenseList);
+        SaveIncome.writeFile(incomeList);
+        SaveBudget.writeFile(budgetEachMonth);
```
I noticed that you only save your data to the file at the end of program. This could be an issue if your program was somehow disrupted during usage and exits unexpectedly. You could consider changing this approach. 👍

jeyvia commented on src/main/java/seedu/brokeMan/entry/EntryList.java:
```diff
@@ -0,0 +1,171 @@
+        }
+        return sum;
+    }
+}
```
These methods are well implemented as well! Well done! 👍

jeyvia commented on src/main/java/seedu/brokeMan/entry/EntryDateComparator.java:
```diff
@@ -0,0 +1,9 @@
+    public int compare(Entry e1, Entry e2) {
+        return e2.time.compareTo(e1.time);
+    }
+}
```
Though not necessary, you could consider just including the two comparators within `EntryList`, since they are each only used once.

jeyvia commented on src/main/java/seedu/brokeMan/parser/StringToCategory.java:
```diff
@@ -0,0 +1,35 @@
+public class StringToCategory {
+    public static Category convertStringToCategory(String categoryString) throws CategoryNotCorrectException {
+        Category category = null;
+        if (categoryString.equals("FOOD")) {
+            category = Category.FOOD;
+        } else if (categoryString.equals("SHOPPING")) {
+            category = Category.SHOPPING;
+        } else if (categoryString.equals("GROCERIES")) {
+            category = Category.GROCERIES;
+        } else if (categoryString.equals("TRANSPORTATION")) {
+            category = Category.TRANSPORTATION;
+        } else if (categoryString.equals("ENTERTAINMENT")) {
+            category = Category.ENTERTAINMENT;
+        } else if (categoryString.equals("TRAVEL")) {
+            category = Category.TRAVEL;
+        } else if (categoryString.equals("SALARY")) {
+            category = Category.SALARY;
+        } else if (categoryString.equals("INVESTMENT")) {
+            category = Category.INVESTMENT;
+        } else if (categoryString.equals("OTHERS")){
+            category = Category.OTHERS;
+        } else {
+            throw new CategoryNotCorrectException();
```
Do consider using constants instead of magic strings here and in other parts of your code as well!

Samueltansw committed:
testing the hyperlink for TOC in user guide

Samueltansw committed:
fixed bugs in table of content links, handled NoSuchElementException input, and some minor view budget, and edit expense commands message usage in consistency

Samueltansw committed:
minor edits to user guide

Samueltansw committed:
updated expected.txt

Samueltansw committed:
Merge pull request #141 from Samueltansw/master

fixed some issues in PED

Samueltansw committed:
edited description of edit an income in user guide

Samueltansw committed:
Merge pull request #142 from Samueltansw/master

edited description of edit an income in user guide

Samueltansw committed:
added limit of 2dp for amount in user guide

Samueltansw committed:
bolded 2dp in user guide

Samueltansw committed:
Merge pull request #143 from Samueltansw/master

added a limit of 2dp for amount in the user guide

Samueltansw committed:
added links to table of content at the end of every section in user guide

Samueltansw committed:
Merge pull request #144 from Samueltansw/master

added links to table of content at the end of every section in user g…

namsengi11 committed:
Update UserGuide.md

Explained that invalid command will call help command

namsengi11 committed:
Update UserGuide.md

Correct instruction for addExpense and addIncome commands.

namsengi11 committed:
Update UserGuide.md

Accounted for one digit months when entering month information for income and expense. 
Ex. 3 instead of 03 for March.

namsengi11 committed:
Update UserGuide.md

Users can remove preceding zeros if apt.

namsengi11 committed:
correct minor issues for command instruction

SpeciLiam committed:
Added exception handling for if the save files are tampered with and have lines that are not correctly readable.

SpeciLiam committed:
Made it to where if the category is tampered with in the save file it does not stop the program, but continues to the next line to read.

namsengi11 committed:
Met test expectations

namsengi11 committed:
resolve issue raised by PE-D

namsengi11 committed:
resolve checkstyle issue

namsengi11 committed:
resolve checkstyle issue

namsengi11 committed:
resolve checkstyle issue

namsengi11 committed:
resolve checkstyle issue

namsengi11 committed:
Merge pull request #145 from namsengi11/test

Fix issues raised by PE-Dry

SpeciLiam committed:
Merge branch 'AY2223S2-CS2113-F13-2:master' into master

SpeciLiam committed:
checkstyle

SpeciLiam committed:
Merge pull request #146 from AY2223S2-CS2113-F13-2/master

Exception handling for tampered save file

namsengi11 committed:
Update UserGuide.md

namsengi11 committed:
Update DeveloperGuide.md

Added implementation information of future features.

namsengi11 committed:
Create namsengi11.md

Samueltansw committed:
added duplicated flags exception for add commands, and also allowed the user to enter '/' character as a description for add commands

Samueltansw committed:
deleted comments to follow coding standard

Samueltansw committed:
updated expected.txt

namsengi11 committed:
Update UserGuide.md

namsengi11 committed:
Update namsengi11.md

Samueltansw committed:
updated SaveExpense and SaveIncome files to accept '/' character as description for add commands

Samueltansw committed:
Merge pull request #147 from Samueltansw/modifyParser

modified parser class

namsengi11 committed:
Add javadoc

Added javadoc description of methods implemented by classes in Entry package.

Samueltansw committed:
disallow user to enter flags as a new description when editing income / expense

Samueltansw committed:
Merge remote-tracking branch 'upstream/master' into modifyParser

Samueltansw committed:
minor edits

Samueltansw committed:
minor edits

Samueltansw committed:
updated Expected.txt

Samueltansw committed:
Merge pull request #149 from Samueltansw/modifyParser

Modify parser

Samueltansw committed:
standardise 'eg.' in command summary in usre guide

Samueltansw committed:
Merge pull request #150 from Samueltansw/modifyParser

standardise 'eg.' in command summary in user guide

Samueltansw committed:
deleted target.java as not in use, and added more JUnit Tests

Samueltansw committed:
Merge pull request #151 from Samueltansw/branch-AddJunitTest2

deleted target.java as not in use, and added more JUnit Tests

Samueltansw committed:
renamed the functions in ParserTest

Samueltansw committed:
minor edits

Samueltansw committed:
Merge pull request #152 from Samueltansw/branch-AddJunitTest2

renamed the functions in ParserTest

Samueltansw committed:
minor edits

Samueltansw committed:
resolve PED issues

sistine-yu committed:
Update part of EntryList component and Parser component in DG

sistine-yu committed:
Add Sistine ppp

sistine-yu committed:
Merge branch 'AY2223S2-CS2113-F13-2:master' into master

Samueltansw committed:
minor bugs

Samueltansw committed:
fixed checkstyle

sistine-yu committed:
Merge pull request #153 from sistine-yu/master

Minor updates in DG

Samueltansw committed:
Merge pull request #154 from Samueltansw/resolvePED

Resolve PED issues

Samueltansw committed:
edited user guide and fixed some bugs

Samueltansw committed:
minor edits to user guide

Samueltansw committed:
updated Expected.txt

Samueltansw committed:
Merge pull request #155 from Samueltansw/master

edit user guide and some PED issues

Samueltansw committed:
minor edits to user guide

Samueltansw committed:
fixed bugs in parser

Samueltansw committed:
Merge pull request #156 from Samueltansw/master

fixed errors in userguide and parser

Samueltansw committed:
updated AboutUs.md and added PPP

Samueltansw committed:
Merge pull request #157 from Samueltansw/master

updated AboutUs.md and added PPP

Samueltansw committed:
edited PPP

Samueltansw committed:
Merge pull request #158 from Samueltansw/master

edited PPP

Samueltansw committed:
added more stuff to PPP

Samueltansw committed:
Merge pull request #161 from Samueltansw/master

added more stuff to PPP

Samueltansw committed:
added more stuff to PPP

Samueltansw committed:
added more stuff to PPP

Samueltansw committed:
fixed checkstyle

Samueltansw committed:
Merge pull request #162 from Samueltansw/master

added more stuff to PPP

Samueltansw committed:
added link for contributions beyond team project

Samueltansw committed:
fixed spelling

Samueltansw committed:
minor edits to PPP

Samueltansw committed:
Merge pull request #163 from Samueltansw/master

minor edits to PPP

Samueltansw committed:
updated README.MD and PPP

sistine-yu committed:
Update the diagram in component part of DG

sistine-yu committed:
Update the component part of DG

sistine-yu committed:
Merge pull request #164 from sistine-yu/master

Update Components parts in DG

sistine-yu committed:
Update PPP

sistine-yu committed:
Merge branch 'AY2223S2-CS2113-F13-2:master' into master

sistine-yu committed:
Merge pull request #165 from sistine-yu/master

Update my ppp

sistine-yu committed:
Minor changes in DG

sistine-yu committed:
Merge remote-tracking branch 'origin/master'

sistine-yu committed:
Merge branch 'AY2223S2-CS2113-F13-2:master' into master

Samueltansw committed:
trying out different links in PPP

Samueltansw committed:
updated the TOC in DG

Samueltansw committed:
updated PPP

Samueltansw committed:
updated the output for help command

sistine-yu committed:
Update readme to display ppp in github pages

sistine-yu committed:
Merge remote-tracking branch 'origin/master'

sistine-yu committed:
Merge pull request #166 from sistine-yu/master

Minor changes in DG

Samueltansw committed:
handled merge conflicts

Samueltansw committed:
minor edits to DG

Samueltansw committed:
fixed merge conflicts

Samueltansw committed:
Merge pull request #167 from Samueltansw/master

updated PPP and UG and DG

Samueltansw committed:
fixed bugs in architecture sequence diagram and added appendix: instructions for manual testing in DG

Samueltansw committed:
updated PPP and removed {more to be added} line in DG

Samueltansw committed:
updated PPP

Samueltansw committed:
Merge pull request #168 from Samueltansw/master

fixed bugs in architecture sequence diagram and added appendix: instr…

Samueltansw committed:
fixed broken link

namsengi11 committed:
Merge pull request #148 from namsengi11/documentation

Add javadoc

Samueltansw committed:
added a maximum limit that can be entered as amount in UG, as well as fixing the decimal issues in BrokeMan

Samueltansw committed:
fixed checkstyle

Samueltansw committed:
updated Expected.txt

Samueltansw committed:
Merge pull request #169 from Samueltansw/master

updated UG and fixed decimal issues in brokeMan

SpeciLiam committed:
Liam Van PPP

SpeciLiam committed:
Merge pull request #170 from AY2223S2-CS2113-F13-2/master

Liam Van PPP

namsengi11 committed:
Add exception for dates smaller than earliest date

Added exception for dates before 2000/01.
Resolved issues raised by PE-D.

namsengi11 committed:
Resolve merge conflict issues

namsengi11 committed:
Resolve test conflicts

namsengi11 committed:
Merge pull request #171 from namsengi11/time

Add exception for dates smaller than earliest date

Samueltansw committed:
modified EntryListClassDiagram, modified descrption of parser component and added parser class diagram and Ui class diagram

Samueltansw committed:
Merge pull request #172 from Samueltansw/master

modified EntryListClassDiagram, modified descrption of parser compone…

Samueltansw committed:
fixed grammar issues and spacing issues, and reorder diagram

Samueltansw committed:
Merge remote-tracking branch 'upstream/master'

namsengi11 committed:
Add tests

namsengi11 committed:
resolve merge conflict

namsengi11 committed:
Merge branch 'master' of https://github.com/namsengi11/tp into test

namsengi11 committed:
Add intensive testing

namsengi11 committed:
Resolve checkstyle issues

namsengi11 committed:
Resolve checkstyle issues

Samueltansw committed:
updated UG and fixed rounding issues in decimal values

Samueltansw committed:
minor edits to UG

namsengi11 committed:
Merge pull request #174 from namsengi11/test

Add intensive tests

Samueltansw committed:
Merge pull request #173 from Samueltansw/master

fixed grammar issues and spacing issues, and reorder diagram

namsengi11 committed:
Update DeveloperGuide.md

Removed private attributes of EntryList as suggested by tutor.

namsengi11 committed:
Update DeveloperGuide.md

Added minor changes suggested by tutor.
Changed word choices.

namsengi11 committed:
Add files via upload

Budget UML

namsengi11 committed:
Delete BudgetClassUML.png

namsengi11 committed:
Add files via upload

namsengi11 committed:
Create budget.puml

namsengi11 committed:
Update DeveloperGuide.md

Added Budget uml.

Samueltansw committed:
edited architecuture diagram, edited and moved the architecture diagram to command component of DG and completed the command component description. Updated PPP

Samueltansw committed:
Merge remote-tracking branch 'upstream/master'

Samueltansw committed:
fixed hyperlink for BudgetClassUML and slight edit to BudgetClassUML.puml

Samueltansw committed:
fixed hyperlink for BudgetClassUML and slight edit to BudgetClassUML.puml

Samueltansw committed:
updated PPP

Samueltansw committed:
Merge pull request #175 from Samueltansw/master

added more items in DG, and updated PPP

Samueltansw committed:
edited DG

Samueltansw committed:
minor change to parser class diagram

Samueltansw committed:
Merge pull request #176 from Samueltansw/master

updated images in DG

Samueltansw committed:
minor edit to PPP

Samueltansw committed:
Merge pull request #177 from Samueltansw/master

minor edit to PPP

Samueltansw committed:
minor edit to PPP and fixed bugs in editing dates in income and expense

Samueltansw committed:
removed commented code

Samueltansw committed:
Merge pull request #178 from Samueltansw/master

fixed bugs in editing time in expense and income

Samueltansw committed:
added oldest date that can be entered in UG

Samueltansw committed:
Merge pull request #179 from Samueltansw/master

added oldest date that can be entered in UG

Samueltansw committed:
added more class diagrams to DG

Samueltansw committed:
Merge pull request #180 from Samueltansw/master

added more class diagrams to DG and updated personal PPP

Samueltansw committed:
updated TOC in DG

Samueltansw committed:
Merge pull request #181 from Samueltansw/master

updated TOC in DG

Samueltansw committed:
edited budget class diagram

Samueltansw committed:
Merge pull request #182 from Samueltansw/master

edited budget class diagram

Samueltansw committed:
minor edits to DG

Samueltansw committed:
Merge pull request #183 from Samueltansw/master

minor edits to DG

Samueltansw committed:
changed the name of .jar file

Samueltansw committed:
Merge pull request #184 from Samueltansw/master

changed the name of .jar file

Samueltansw committed:
fixed bugs in setBudget command

Samueltansw committed:
removed commented code

Samueltansw committed:
fixed bugs in parser test

Samueltansw committed:
Merge pull request #185 from Samueltansw/master

fixed bugs in set budget command

namsengi11 committed:
Update DeveloperGuide.md

Added details for class implementation.

namsengi11 committed:
Update DeveloperGuide.md

Samueltansw committed:
added samuel's image and updated aboutus.md

namsengi11 committed:
Add logging and reduce repetitive testing

Samueltansw committed:
Merge pull request #187 from Samueltansw/master

added samuel's image and updated aboutus.md

namsengi11 committed:
Merge pull request #186 from namsengi11/test

Add logging and reduce repetitive testing

Samueltansw committed:
updated UG

namsengi11 committed:
Delete D.jpg

Samueltansw committed:
Merge remote-tracking branch 'upstream/master'

namsengi11 committed:
Add files via upload

namsengi11 committed:
Update AboutUs.md

Samueltansw committed:
Merge pull request #188 from Samueltansw/master

fixed UG flaw

namsengi11 committed:
Update README.md

Copied readme also to the project root.

namsengi11 committed:
Update README.md

namsengi11 committed:
Update README.md

namsengi11 committed:
Update DeveloperGuide.md

namsengi11 committed:
Add date limit

namsengi11 committed:
Merge branch 'master' of https://github.com/namsengi11/tp into test

namsengi11 committed:
Update date exception message

namsengi11 committed:
Updated tests

namsengi11 committed:
Merge pull request #189 from namsengi11/test

Test

Samueltansw committed:
updated UG

Samueltansw committed:
Merge remote-tracking branch 'upstream/master'

Samueltansw committed:
Merge pull request #190 from Samueltansw/master

updated UG

namsengi11 committed:
Update namsengi11.md

namsengi11 committed:
Update namsengi11.md

namsengi11 committed:
Update namsengi11.md
